app/drivers/kscan/debounce.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <sys/util.h>
+
+#define DEBOUNCE_COUNTER_BITS 14
+#define DEBOUNCE_COUNTER_MAX BIT_MASK(DEBOUNCE_COUNTER_BITS)
+
+struct debounce_state {
```

Comment:
Are you trying to pack this to avoid extra storage overhead?

Reply:
Correct. I figured I might as well use as little space as possible for the best chance of being able to run things on cheaper processors.